utils/dataconvert/dataconvert.cpp

```diff
@@ -3005,6 +3005,14 @@ int64_t DataConvert::intToDatetime(int64_t data, bool* date)
     else
     {
         y = atoi(year.c_str());
+
+        // special handling for 2-byte year
+        if (year.length() == 2)
```

Comment:
Found a bug while going through a test case. Year can be in format YY so this is needed so isDateValid does not flag this year as invalid. These magic numbers are taken from function mysql_str_to_datetime lines 711-714.